### PR TITLE
Fix #2544 proprietary_sim_types added to sim_types

### LINKED ISSUE #2544: add values from proprietary_sim_types to sim_types
proprietary_sim_types should be unioned with sim_types so that any proprietary_sim_types is guaranteed in sim_types.  This makes configuration management easier.

default proprietary sim types can include flash for dev, but not alpha, etc.

### PR DESCRIPTION
**Pull request by robnagler**
sim_types is now a frozenset

### CODE REVIEW

**Comment by robnagler** [comment]
Actually hold off. Need to validate types.

**Comment by robnagler** [comment]
@e-carlin ready

File: sirepo/feature_config.py [hunk 1]
```diff
@@ -111,4 +99,15 @@ def _codes():
             display_test_boxes=(pkconfig.channel_in_internal_test(), bool, 'Display test boxes to visualize 3D -> 2D projections'),
         ),
     )
+    a = ALL_CODES if pkconfig.channel_in_internal_test() else NON_ALPHA_CODES
+    s = set(_cfg.sim_types) if _cfg.sim_types else set(a)
+    if _cfg.proprietary_sim_types:
+        s.update(_cfg.proprietary_sim_types)
+    # need more cases like this
+    if 'jspec' in s and 'elegant' not in s:
```
Comment (e-carlin) [inline sirepo/feature_config.py:107]: Why check explicitly for elegant?

File: sirepo/pkcli/admin.py [hunk 2]
```diff
@@ -99,13 +99,12 @@ def setup_dev_proprietary_code(sim_type, rpm_url):
       sim_type (str): simulation type
       rpm_url (str): Url of the rpm (file:// or http://)
     """
-    import sirepo.pkcli.admin
     import urllib.request
 
     assert pkconfig.channel_in('dev'), \
         'Only to be used in dev. channel={}'.format(pkconfig.cfg.channel)
 
-    d = sirepo.pkcli.admin.proprietary_code_dir(sim_type)
```
Comment (e-carlin) [inline sirepo/pkcli/admin.py:108]: Geez, who wrote that code!

File: sirepo/feature_config.py [hunk 3]
```diff
@@ -111,4 +99,15 @@ def _codes():
             display_test_boxes=(pkconfig.channel_in_internal_test(), bool, 'Display test boxes to visualize 3D -> 2D projections'),
         ),
     )
+    a = ALL_CODES if pkconfig.channel_in_internal_test() else NON_ALPHA_CODES
+    s = set(_cfg.sim_types) if _cfg.sim_types else set(a)
+    if _cfg.proprietary_sim_types:
```
Comment (e-carlin) [inline sirepo/feature_config.py:104]: Why check?

File: sirepo/feature_config.py (hunk 1 above)
Comment (robnagler) [reply sirepo/feature_config.py:108]: New comment:

```py
    # jspec imports elegant, but elegant won't work if it is not a valid
    # sim_type so need to include here. Need a better model of
    # dependencies between codes.
```

File: sirepo/feature_config.py (hunk 3 above)
Comment (robnagler) [reply sirepo/feature_config.py]: removed. old code was frozenset so was complicated to udpate.

File: sirepo/pkcli/admin.py (hunk 2 above)
Comment (robnagler) [reply sirepo/pkcli/admin.py:108]: As Buffett said "I've made so many mistakes, if I tried to kick myself, my legs would be exhausted."

**Pull request merged**